Settings/UserSettings.cs

```diff
@@ -117,6 +117,9 @@ public sealed class Config
         public string ExtraArgs { get; set; }
 
         [XmlElement("LauncherVersion")]
+        public String LocalLauncherVersionS { get; set; }
+
+        [XmlIgnore]
```

Comment:
Why are there now two properties, and what does the S denote?

Reply:
LocalLauncherVersionS is a string converted from LocalLauncherVersion.
The S denote String.
These two properties should keep same. So I name it as that.
(Do you have any better name suggested?)

It seems that Version type could not be serialized and saved in the XML files. But it can be saved by coverting it into a String.
And, old format of LauncherVersion can't be parse to Version type directly. Adding a String can store this old format of version, and make it easy to transform it into new format.

In order not affecting other part of code and the struct of config file, I add this string at here, which will be saved and 
loaded automaticly by serializer. 
Once it was loaded, it would be checked and assigned to the original LauncherVersion. 
This properties has no affect in other parts of program.

Reply:
I would call it LocalLauncherStringVersion. Additionally, you should make so `public String` becomes `public string` instead.

Reply:
Oops. 
It's my mistake.